dbms/src/Core/Block.cpp

```diff
@@ -514,6 +514,26 @@ static ReturnType checkBlockStructure(const Block & lhs, const Block & rhs, cons
     return ReturnType(true);
 }
 
+Block mergeBlocks(Blocks && result_blocks)
+{
+    auto & sample_block = result_blocks[0];
+    MutableColumns dst_columns(sample_block.columns());
+    for (size_t i = 0; i < sample_block.columns(); i++)
+    {
+        dst_columns[i] = sample_block.getByPosition(i).column->cloneEmpty();
```

Comment:
why assigns to dst_columns again here?

Reply:
> why assigns to dst_columns again here?

dst_columns is a vector, if not assign it here the point of column in the vector dst_columns will be not initialized，and the code ```dst_columns[column]->insertRangeFrom(XXX)``` below will throw exception.

Reply:
I mean, in this way, the first block's memory is not reused any more.

Reply:
> I mean, in this way, the first block's memory is not reused any more.

done.